Resources/public/js/views/services/plugins/ez-objectrelationsloadplugin.js

```diff
@@ -64,11 +59,78 @@ YUI.add('ez-objectrelationsloadplugin', function (Y) {
 
             stack.done(function () {
                 e.target.setAttrs({
-                    relatedContents: relatedContentListArray,
+                    relatedContents: relatedContentStructListArray,
                     loadingError: loadingError,
                 });
             });
         },
+
+        /**
+         * Loads content struct containing content and main location for given content id.
+         * The contentStruct is being passed to the callback function.
+         *
+         * @method _loadContentStruct
+         * @protected
+         * @param {String} contentId
+         * @param {Function} callback
+         * @param {Bool} callback.error
+         * @param {Object} callback.contentStruct content struct with loaded content and location
+         * @param {eZ.Content} callback.contentStruct.content
+         * @param {eZ.Location} callback.contentStruct.location
+         */
+        _loadContentStruct: function (contentId, callback) {
+            var ContentModel = this.get('contentModelConstructor'),
+                content = new ContentModel(),
+                loadOptions = {api: this.get('host').get('capi')},
+                that = this,
+                contentStruct;
+
+            content.set('id', contentId);
+            content.load(loadOptions, function (error) {
+                if (error) {
+                    callback(error);
+                } else {
+                    contentStruct = {content: content};
+                    that._loadLocationToContentStruct(contentStruct, callback);
```

Comment:
loading the Location and loading the Location's path should be triggered by a flag in the event otherwise, we'll do that everytime while this is not needed for instance when displaying the relation when editing a content.